File: 0344. Reverse String/Solution04.py
/*
Submission Detail:{
    Difficulty : Easy
    Acceptance Rate : 75.90 %
    Runtime : 300 ms
    Memory Usage : 21.1 MB
    Testcase : 477 / 477 passed
    Ranking : 
        Your runtime beats 30.65 % of python submissions.
        Your memory usage beats 78.95 % of python submissions.
}
*/

class Solution:
    def reverseString(self, s):
        left, right = 0, len(s) - 1
        while left < right:
            s[left], s[right] = s[right], s[left]
            left, right = left + 1, right - 1

File: 0792. Binary Search/Solution04.java
/*
Submission Detail:{
    Difficulty : Easy
    Acceptance Rate : 55.14 %
    Runtime : 0 ms
    Memory Usage : 54.3 MB
    Testcase : 47 / 47 passed
    Ranking : 
        Your runtime beats 100.00 % of java submissions.
        Your memory usage beats 53.31 % of java submissions.
}
*/

class Solution {
    public int search(int[] nums, int target) {
    int left=0;
    int right=nums.length-1;
        while(left<=right){
            int mid= left+(right-left)/2;
            if(nums[mid]==target){
                return mid;
            }
            else if(nums[mid]>target){
                right=mid-1;
                
            }
            else if(nums[mid]<target){
                left=mid+1;
            }
        }
        return -1;
    }
}